librz/bin/format/le/le.c

```diff
@@ -57,828 +87,1749 @@ const char *le_get_arch(rz_bin_le_obj_t *bin) {
 	}
 }
 
-static char *le_read_nonnull_str_at(RzBuffer *buf, ut64 *offset) {
-	ut8 size;
-	if (!rz_buf_read8_at(buf, *offset, &size)) {
-		return NULL;
+static bool le_read_len_str_offset(RzBuffer *buf, ut64 *offset, char **out) {
+	*out = NULL;
+	ut8 len;
+	if (!rz_buf_read8_offset(buf, offset, &len)) {
+		return false;
+	}
+
+	if (!len) {
+		return true; // success yet *out == NULL, this is why the return value is bool
+	}
+
+	ut8 *str = calloc((size_t)len + 1, sizeof(char));
+	if (!str) {
+		return false;
+	}
+	if (!rz_buf_read_offset(buf, offset, str, len)) {
+		free(str);
+		return false;
+	}
+	for (ut8 *s = str; s != str + len; ++s) {
```

Comment:
For strings we might have better functions, not sure if it's appropriate to use them here though. cc @wargio

Reply:
depends if this is only used here or elsewhere.